hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/cache/EntityMapWarmer.java

```diff
@@ -0,0 +1,258 @@
+            ThreadPoolExecutor threadpool) {
+        this.accountsStorageAdapter = accountsStorageAdapter;
+        this.nftsAdapter = nftsAdapter;
+        this.tokenRelsAdapter = tokenRelsAdapter;
+        this.threadpool = threadpool;
+    }
+
+    public static EntityMapWarmer getInstance(
+            Supplier<AccountStorageAdapter> accountsStorageAdapter,
+            Supplier<UniqueTokenMapAdapter> nftsAdapter,
+            Supplier<TokenRelStorageAdapter> tokenRelsAdapter,
+            GlobalDynamicProperties globalDynamicProperties) {
+        if (instance == null) {
+            instance =
+                    new EntityMapWarmer(accountsStorageAdapter, nftsAdapter, tokenRelsAdapter, globalDynamicProperties);
+        }
+        return instance;
+    }
+
+    /**
+     * Warms the cache for select entities found in the given {@code Round}. Any entities not
+     * found will, of course, not be warmed, but instead will essentially be ignored.
+     *
+     * @param round the round to warm the cache for
+     */
+    public void warmCache(Round round) {
+        removeAllTasks();
+        threadpool.execute(() -> doWarmups(round));
+    }
+
+    // "warmTokenObjs" as in "warm the token and its associated token relation objects"
+    // for both the sending and receiving accounts
+    private void warmTokenObjs(AccountID sender, AccountID receiver, long tokenNum, long senderSerialNum) {
+        if (accountsStorageAdapter.get().areOnDisk()
+                && nftsAdapter.get().isVirtual()
+                && tokenRelsAdapter.get().areOnDisk()) {
+
+            threadpool.execute(() -> {
+                // Sender:
+                final var acctMap = accountsStorageAdapter.get().getOnDiskAccounts();
+                final OnDiskAccount account =
+                        Objects.requireNonNull(acctMap).get(EntityNumVirtualKey.from(EntityNum.fromAccountId(sender)));
+                if (account != null) {
+                    final var headNum = account.getHeadNftTokenNum();
+                    if (headNum != 0) {
+                        final var headSerialNum = account.getHeadNftSerialNum();
+                        final var nftMap = nftsAdapter.get().getOnDiskNfts();
+                        Objects.requireNonNull(nftMap).warm(new UniqueTokenKey(headNum, headSerialNum));
+                    }
+                }
+            });
+
+            threadpool.execute(() -> {
+                // Receiver:
+                final var acctMap = accountsStorageAdapter.get().getOnDiskAccounts();
+                final OnDiskAccount account = Objects.requireNonNull(acctMap)
+                        .get(EntityNumVirtualKey.from(EntityNum.fromAccountId(receiver)));
+                if (account != null) {
+                    final var headTokenId = account.getHeadTokenId();
+                    if (headTokenId != 0) {
+                        final var tokenRelMap = tokenRelsAdapter.get().getOnDiskRels();
```

Comment:
Are the benchmarks using a lot of auto-association slots? In general this would be wasted work for a `CryptoTransfer`. 

If it's no big deal to make a superfluous `warm()` call, probably doesn't matter.

Reply:
Yes, the benchmark is designed to use auto-association allowing an NFT to be sent to a random account from a predetermined subset of 1000 accounts "associated" with that token. Explicitly associating, say, 10M accounts with 1000 tokens each can accept was considered impractical (would take too long).
From a different angle, auto-association incurs more overhead during transaction handling and thus is more preferable for benchmarking.
Superfluous `warm()` calls are acceptable according to current measurement but that should be monitored and potentially reevaluated in the future.

Reply:
If I read this right, @OlegMazurov you're saying that we do need this code since we do want to use auto-association for benchmarking? Is this concern resolved?

Reply:
I would like to keep this code, yes.